src/main.ts

```diff
@@ -363,23 +369,28 @@ async function run_linux(): Promise<void> {
     }
 
     await io.rmRF(cache_dir)
+    return true
   } catch (error: unknown) {
     if (error instanceof Error) {
       core.setFailed(error.message)
     } else {
       core.setFailed(`No error message`)
     }
   }
+  return false
 }
 
 export async function run(): Promise<void> {
+  let success = false
   if (process.platform === 'linux') {
-    await run_linux()
+    success = await run_linux()
   } else {
     core.setFailed(`Action doesn't support ${process.platform} platform`)
   }
 
-  await exec.exec('tarantool --version')
+  if (success) {
+    await exec.exec('tarantool --version')
+  }
```

Comment:
Wouldn't it be clearer, if we enclose the `run_linux()` call in a `try-catch` block and just move the `tarantool --version` call into that block?

Reply:
The `run_linux()` function already has a try-catch block inside. I can move it outside.

```diff
diff --git a/src/main.ts b/src/main.ts
index 0526d6d..8c10824 100644
--- a/src/main.ts
+++ b/src/main.ts
@@ -268,7 +268,6 @@ function dpkg_is_file_included(
 }
 
 async function run_linux(): Promise<void> {
-  try {
   const distro = await lsb_release()
   const distro_id = (await lsb_release_id()).toLowerCase()
   const cache_dir = 'cache-tarantool'
@@ -369,23 +368,25 @@ async function run_linux(): Promise<void> {
   }
 
   await io.rmRF(cache_dir)
-  } catch (error: unknown) {
-    if (error instanceof Error) {
-      core.setFailed(error.message)
-    } else {
-      core.setFailed(`No error message`)
-    }
-  }
 }
 
 export async function run(): Promise<void> {
-  if (process.platform === 'linux') {
-    await run_linux()
-  } else {
+  if (process.platform !== 'linux') {
     core.setFailed(`Action doesn't support ${process.platform} platform`)
+    return
   }
 
-  await exec.exec('tarantool --version')
+  await run_linux()
+    .then(_ => {
+      return exec.exec('tarantool --version')
+    })
+    .catch(error => {
+      if (error instanceof Error) {
+        core.setFailed(error.message)
+      } else {
+        core.setFailed(`No error message`)
+      }
+    })
 }
 
 // Export core.setOutput() to use in testing of setup-tarantool.
```

(The diff is created with `-b` to skip indent changes.)

Does it look better?

Reply:
Yes! Now the `run_linux()` call and it's error handling is done in one place. And the platform compatibility check is separate.

Reply:
OK, I've updated it this way.